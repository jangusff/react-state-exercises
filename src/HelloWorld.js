import React, { Component } from 'react';
import TheDate from './state/TheDate';

class HelloWorld extends React.Component {

  static defaultProps = {
    step: 1
  };

  constructor(props) {
    //console.log('props in constructor', props);
    super(props);
    this.state = { who: "World" };
    // this.handleButtonClick = this.handleButtonClick.bind(this); <-- one option to get around React's "TypeError: Cannot read property 'state' of undefined".
  }

 
  render() {
      /* <button onClick={function() { console.log('clicked!') }}> */
      /* <button onClick={this.handleButtonClick}> */
      /* <button onClick={() => this.handleButtonClick()}>  <-- another option to get around React's "TypeError: Cannot read property 'state' of undefined". */
      return (
        <div className='HelloWorld'>
        <p>Hello, {this.state.who}!</p>
        <button
          onClick={() => this.setState({ who: 'world' })}
        >
          World
        </button>
        <br/>
        <button
          onClick={() => this.setState({ who: 'friend' })}
        >
          Friend
        </button>
        <br />
        <button
          onClick={() => this.setState({ who: 'React' })}
        >
          React
        </button>
      </div>
    )
  }
}

export default HelloWorld;

