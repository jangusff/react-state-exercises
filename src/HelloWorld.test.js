import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import HelloWorld from './HelloWorld'

describe(`HelloWorld Component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HelloWorld />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  
  //it.skip('renders the UI as expected', () => {
  it('renders the UI as expected', () => {
    expect(
      renderer.create(<HelloWorld />).toJSON()
    ).toMatchSnapshot()
  })
})