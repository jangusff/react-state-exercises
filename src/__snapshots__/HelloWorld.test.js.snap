// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld Component renders the UI as expected 1`] = `
<div>
  <p>
    Hello, 
    world
  </p>
  <button
    onClick={[Function]}
  >
    World
  </button>
  <button
    onClick={[Function]}
  >
    Friend
  </button>
  <button
    onClick={[Function]}
  >
    React
  </button>
</div>
`;
